### PR TITLE
feat: [CI-6037]: Implement containerless plugin step

### PR DESCRIPTION
**Pull request by shubham149**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39683)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java [hunk 1]
```diff
@@ -64,12 +72,28 @@ public VmPluginStep serialize(PluginStepInfo pluginStepInfo, StageInfraDetails s
...
         "connectorRef", stepName, identifier, pluginStepInfo.getConnectorRef(), false);
-
     String image =
         RunTimeInputHandler.resolveStringParameter("Image", stepName, identifier, pluginStepInfo.getImage(), false);
-
+    String uses =
+        RunTimeInputHandler.resolveStringParameter("uses", stepName, identifier, pluginStepInfo.getUses(), false);
     long timeout = TimeoutUtils.getTimeoutInSeconds(parameterFieldTimeout, pluginStepInfo.getDefaultTimeout());
 
+    if (isNotEmpty(image)) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java:81]: Do you also want make sure uses is empty ?

Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java:81]: What if both uses and command exists ?

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java [hunk 2]
```diff
@@ -47,7 +55,7 @@ public class VmPluginStepSerializer {
   @Inject ConnectorUtils connectorUtils;
   @Inject HarnessImageUtils harnessImageUtils;
 
-  public VmPluginStep serialize(PluginStepInfo pluginStepInfo, StageInfraDetails stageInfraDetails, String identifier,
+  public VmStepInfo serialize(PluginStepInfo pluginStepInfo, StageInfraDetails stageInfraDetails, String identifier,
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java:58]: Is it a contract change ?

Comment (shubham149) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java:58]: No

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java (hunk 1 above)
Comment (shubham149) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmPluginStepSerializer.java:81]: Add a log warning

**Pull request merged**